fhir-server-spi/src/main/java/com/ibm/fhir/server/spi/operation/FHIROperationUtil.java

```diff
@@ -72,103 +73,124 @@ public static void init() {
-                                FHIROperationException operationException =
-                                        new FHIROperationException("Query parameter '" + name + "' is an invalid type: '" + typeName + "'");
-
-                                List<Issue> issues = new ArrayList<>();
-                                Issue.Builder builder = Issue.builder();
-                                builder.code(IssueType.INVALID);
-                                builder.diagnostics(string("Query parameter '" + name + "' is an invalid type: '" + typeName + "'"));
-                                builder.severity(IssueSeverity.ERROR);
-                                issues.add(builder.build());
-
-                                operationException.setIssues(issues);
-
-                                throw operationException;
-                            }
-                            parametersBuilder.parameter(parameterBuilder.build());
-                        }
+
+            for (OperationDefinition.Parameter parameter : definition.getParameter()) {
+                if (OperationParameterUse.IN.getValue().equals(parameter.getUse().getValue())) {
+                    String paramName = parameter.getName().getValue();
+                    if (!queryParameters.containsKey(paramName)) {
+                        continue;
+                    }
+
+                    FHIRAllTypes type = parameter.getType();
+                    if (type == null) {
+                        continue;
+                    }
+                    String typeName = type.getValue();
+
+                    for (String value : queryParameters.get(paramName)) {
+                        parametersBuilder.parameter(parseParameter(typeName, paramName, value));
                     }
                 }
             }
             return parametersBuilder.build();
         } catch (FHIROperationException e) {
             throw e;
         } catch (Exception e) {
-            // Originally returned 500 when it should be 400 (it's on the client).
-            FHIROperationException operationException =
-                    new FHIROperationException("Unable to process query parameters", e);
+            throw new FHIROperationException("Unable to process query parameters", e);
+        }
+    }
+
+    private static Parameter parseParameter(String typeName, String paramName, String value) throws FHIROperationException {
 
-            List<Issue> issues = new ArrayList<>();
-            Issue.Builder builder = Issue.builder();
-            builder.code(IssueType.INVALID);
-            builder.diagnostics(string("Unable to process query parameters"));
-            builder.severity(IssueSeverity.ERROR);
-            issues.add(builder.build());
+        Parameter.Builder parameterBuilder = Parameter.builder().name(paramName);
+        try {
+            if ("boolean".equals(typeName)) {
```

Comment:
in this PR I mostly just split this long if else block into its own method and tightened up the error handling a bit (and added tests).  personally i think its ok, but agree there might be cleaner approaches

Reply:
one thing we could do without much trouble is change it into a switch over the FHIRAllTypes.Value enum.

the only reason I didn't do that is that this enum has A LOT of values and we really only care about these 16 or so primitive ones.  so i think it would be about the same in practice.